### ISSUE TITLE
Exception classes should all extend a common botocore exception

### ISSUE THREAD

**Issue by marcogallotta** [boto/botocore#899]
Right now, if I want to handle all classes of botocore exceptions, I can't easily do so. For example, `ClientError` extends `Exception` directly. If they all extended from `BotoCoreError` or something, then I could easily handle any botocore exception.

**Comment by jamesls** [comment]
I agree that this would be nice, I think the intent was for `ClientError` to subclass from `BotoCoreError`.  I think there's a small chance of backwards incompatibility with something like:

```
try:
    client.make_api_call()
except BotoCoreError:
    handle_botocore_error()
except ClientError:
    handle_client_error()
```

In this case the `ClientError` case would now no longer be executed.

Thoughts? cc @kyleknap @JordonPhillips

**Comment by kyleknap** [comment]
I did not know that ClientError was subclassed directly from Exception. I would have expected it to subclass from BotoCoreError in the first place. There is definitely a chance for backwards incompatibilities here, but I think it is small enough to be a non-issue.

**Comment by kyleknap** [comment]
I am actually leaning toward calling it a bug even as that is not the intent for ClientError to not subclass from BotoCoreError.

**Comment by thehesiod** [comment]
hopefully changes for this could incorporate issue raised by https://github.com/boto/botocore/issues/834

**Comment by joguSD** [comment]
The exceptions in [botocore/models.py](https://github.com/boto/botocore/blob/develop/botocore/model.py) also extend `Exception` directly.
